### PR TITLE
 feat(cat): implement an additional level for --info to cat command, plus additional validations to files and nrs commands

### PR DESCRIPTION
**Pull request by bochaco**
- Add a validation to files and nrs commands and APIs, to make sure NRS name provided to nrs create/add, as well as target URL provided to files sync commands are unversioned
- Implementation of an additional level for the `--info` argument in the `cat` command to display the content of the NRS map, e.g.:
```
Resolved using NRS Map:
PublicName: "mypubname"
Container XOR-URL: safe://hnyydyiec84yks653p87zycggzyh691a8y49n5n7srh4awjpzaayhfgnunbqh
Native data type: PublishedSeqAppendOnlyData
Type tag: 1500
XOR name: 0x50c3e80ab7b7969fb7030c6b839efcb0706be2d8bb627358a25b7c601c298531
Version: 4
+------------------------+----------------------+----------------------+--------------------------------------------------------------------------+
| NRS name/subname       | Created              | Modified             | Link                                                                     |
+------------------------+----------------------+----------------------+--------------------------------------------------------------------------+
| mypubname              | 2019-08-12T15:28:16Z | 2019-08-12T15:28:16Z | safe://hnyynysfouck3wgmssrt75nixhrdf47skzo86wfbx91uzjh46aqt6mzn6nbnc?v=0 |
+------------------------+----------------------+----------------------+--------------------------------------------------------------------------+
| subname.mypubname      | 2019-08-12T15:29:42Z | 2019-08-12T15:29:42Z | safe://hnyynyi37x4gjs7y815y3q8dmkaugnrf3ma4ezuehr3adzibgo116o6zehbnc?v=0 |
+------------------------+----------------------+----------------------+--------------------------------------------------------------------------+
| test.subname.mypubname | 2019-08-12T15:30:00Z | 2019-08-12T15:30:00Z | safe://hbyyyydisscu7ynfxnyknqmi94bcikkn4d1tnfsadbzu8fgktb8rdn1h6a        |
+------------------------+----------------------+----------------------+--------------------------------------------------------------------------+
```

### CODE REVIEW

File: src/api/nrs_map.rs [hunk 1]
```diff
@@ -302,8 +309,8 @@ fn validate_nrs_link(link: &str) -> ResultReturn<()> {
         match link_encoder.content_type() {
             SafeContentType::FilesContainer | SafeContentType::NrsMapContainer => {
                 Err(Error::InvalidInput(format!(
-                    "The link is unversioned, but the linked content is versionable. NRS resolver doesn\'t allow unversioned links for this type of content: \"{}\"",
-                    link
+                    "The link is unversioned, but the linked content ({}) is versionable. NRS resolver doesn\'t allow unversioned links for this type of content: \"{}\"",
```
Comment (joshuef) [inline src/api/nrs_map.rs]: Perhaps we need to note that if it _can_ be versioned, it _must_ be versioned?

Comment (bochaco) [reply src/api/nrs_map.rs]: Yes, I had a hard time wtiting that error message :D , let me try to come up with another version of it @joshuef

**Comment by calumcraig** [comment]
Tested command `nrs create`, `nrs add` and `files sync` with versioned and non versioned <xorurl>s:

`nrs create`
```
> ./safe nrs create testfolder20 --link safe://hnyynys5cgnfy5eh5f9y7orf811mnnswokh9wt1skkfx6abyyfzszo5yf4bnc
[2019-08-20T08:05:50Z ERROR safe] safe_cli error: [Error] InvalidInput - The linked content (FilesContainer) is versionable, therefore NRS requires the link to specify a version: "safe://hnyynys5cgnfy5eh5f9y7orf811mnnswokh9wt1skkfx6abyyfzszo5yf4bnc"
```
>>> fails
Add version tag to <xorurl>
```
> ./safe nrs create testfolder20 --link "safe://hnyynys5cgnfy5eh5f9y7orf811mnnswokh9wt1skkfx6abyyfzszo5yf4bnc?v=1"
New NRS Map for "safe://testfolder20" created at: "safe://hnyydyiaxhpcbfjpjtn1baji846f169f8qz34aes48dhfpmitqe7en4htrbqh"
+  testfolder20  safe://hnyynys5cgnfy5eh5f9y7orf811mnnswokh9wt1skkfx6abyyfzszo5yf4bnc?v=1 
```
>>> success

**Comment by calumcraig** [comment]
`nrs add`
```
> ./safe nrs add testfolder20 --link safe://hnyynys5cgnfy5eh5f9y7orf811mnnswokh9wt1skkfx6abyyfzszo5yf4bnc
[2019-08-20T08:34:25Z ERROR safe] safe_cli error: [Error] InvalidInput - The linked content (FilesContainer) is versionable, therefore NRS requires the link to specify a version: "safe://hnyynys5cgnfy5eh5f9y7orf811mnnswokh9wt1skkfx6abyyfzszo5yf4bnc"
```
- fails

Add version tag to the xorurl:
```
> ./safe nrs add testfolder20 --link "safe://hnyynys5cgnfy5eh5f9y7orf811mnnswokh9wt1skkfx6abyyfzszo5yf4bnc?v=0"

NRS Map updated (version 1): "safe://hnyydyiaxhpcbfjpjtn1baji846f169f8qz34aes48dhfpmitqe7en4htrbqh"
+  testfolder20  safe://hnyynys5cgnfy5eh5f9y7orf811mnnswokh9wt1skkfx6abyyfzszo5yf4bnc?v=0
```
- success

**Comment by calumcraig** [comment]
`files sync`
```
> ./safe files sync testfolder/ "safe://hnyynys5cgnfy5eh5f9y7orf811mnnswokh9wt1skkfx6abyyfzszo5yf4bnc?v=0" -r -d
[2019-08-20T12:06:53Z ERROR safe] safe_cli error: [Error] InvalidInput - The target URL cannot cannot contain a version: safe://hnyynys5cgnfy5eh5f9y7orf811mnnswokh9wt1skkfx6abyyfzszo5yf4bnc?v=0
```
- fails

Add version tag to the xorurlL
```
> ./safe files sync testfolder/ safe://hnyynys5cgnfy5eh5f9y7orf811mnnswokh9wt1skkfx6abyyfzszo5yf4bnc -r -d
FilesContainer synced up (version 1): "safe://hnyynys5cgnfy5eh5f9y7orf811mnnswokh9wt1skkfx6abyyfzszo5yf4bnc"
-  /image1.png             safe://hbyyyynmimtuwcq4kbttgt14pbdk3amf7wr9a1cah8ounx953o1zx4gxoj 
+  testfolder/newinfo.txt  safe://hbyyyydj99a5hm68szc3ehnt4swbo7caziod9w53b5j87efsykjqyxoo4k 
*  testfolder/text1.txt    safe://hbyyyydntn8o4o3mtqw3z1ewtuydsqgmpagm57zr3pbrdtdzwmj6y67g6a 
```
- success

**Comment by calumcraig** [comment]
Testing the additional info displayed for `safe cat --info`:
Using xorurls:
```
> ./safe cat safe://hnyynyzc5j3abzbkaam8jnsw84j15if1fkoordp7a53wsj889asww3zgyhbnc
Files of FilesContainer (version 0) at "safe://hnyynyzc5j3abzbkaam8jnsw84j15if1fkoordp7a53wsj889asww3zgyhbnc":
+----------------+--------+----------------------+----------------------+-------------------------------------------------------------------+
| Name           | Size   | Created              | Modified             | Link                                                              |
+----------------+--------+----------------------+----------------------+-------------------------------------------------------------------+
| /image1.png    | 102400 | 2019-08-20T12:14:26Z | 2019-08-20T12:14:26Z | safe://hbyyyynmimtuwcq4kbttgt14pbdk3amf7wr9a1cah8ounx953o1zx4gxoj |
+----------------+--------+----------------------+----------------------+-------------------------------------------------------------------+
| /text1.txt     | 102400 | 2019-08-20T12:14:26Z | 2019-08-20T12:14:26Z | safe://hbyyyynmimtuwcq4kbttgt14pbdk3amf7wr9a1cah8ounx953o1zx4gxoj |
+----------------+--------+----------------------+----------------------+-------------------------------------------------------------------+
| /webpage1.html | 102400 | 2019-08-20T12:14:26Z | 2019-08-20T12:14:26Z | safe://hbyyyynmimtuwcq4kbttgt14pbdk3amf7wr9a1cah8ounx953o1zx4gxoj |
+----------------+--------+----------------------+----------------------+-------------------------------------------------------------------+
```
```
> ./safe cat safe://hnyynyzc5j3abzbkaam8jnsw84j15if1fkoordp7a53wsj889asww3zgyhbnc --info
Native data type: PublishedSeqAppendOnlyData
Type tag: 1100
XOR name: 0xd9b4e701b8558c2ce915a87d265ba9645542041b7b8de69649cffc5a94cdcc0e


Files of FilesContainer (version 0) at "safe://hnyynyzc5j3abzbkaam8jnsw84j15if1fkoordp7a53wsj889asww3zgyhbnc":
+----------------+--------+----------------------+----------------------+-------------------------------------------------------------------+
| Name           | Size   | Created              | Modified             | Link                                                              |
+----------------+--------+----------------------+----------------------+-------------------------------------------------------------------+
| /image1.png    | 102400 | 2019-08-20T12:14:26Z | 2019-08-20T12:14:26Z | safe://hbyyyynmimtuwcq4kbttgt14pbdk3amf7wr9a1cah8ounx953o1zx4gxoj |
+----------------+--------+----------------------+----------------------+-------------------------------------------------------------------+
| /text1.txt     | 102400 | 2019-08-20T12:14:26Z | 2019-08-20T12:14:26Z | safe://hbyyyynmimtuwcq4kbttgt14pbdk3amf7wr9a1cah8ounx953o1zx4gxoj |
+----------------+--------+----------------------+----------------------+-------------------------------------------------------------------+
| /webpage1.html | 102400 | 2019-08-20T12:14:26Z | 2019-08-20T12:14:26Z | safe://hbyyyynmimtuwcq4kbttgt14pbdk3amf7wr9a1cah8ounx953o1zx4gxoj |
+----------------+--------+----------------------+----------------------+-------------------------------------------------------------------+
```

**Comment by calumcraig** [comment]
Using nrsurls:
```
> ./safe cat safe://testfolder200820192
Files of FilesContainer (version 0) at "safe://testfolder200820192":
+----------------+--------+----------------------+----------------------+-------------------------------------------------------------------+
| Name           | Size   | Created              | Modified             | Link                                                              |
+----------------+--------+----------------------+----------------------+-------------------------------------------------------------------+
| /image1.png    | 102400 | 2019-08-20T12:14:26Z | 2019-08-20T12:14:26Z | safe://hbyyyynmimtuwcq4kbttgt14pbdk3amf7wr9a1cah8ounx953o1zx4gxoj |
+----------------+--------+----------------------+----------------------+-------------------------------------------------------------------+
| /text1.txt     | 102400 | 2019-08-20T12:14:26Z | 2019-08-20T12:14:26Z | safe://hbyyyynmimtuwcq4kbttgt14pbdk3amf7wr9a1cah8ounx953o1zx4gxoj |
+----------------+--------+----------------------+----------------------+-------------------------------------------------------------------+
| /webpage1.html | 102400 | 2019-08-20T12:14:26Z | 2019-08-20T12:14:26Z | safe://hbyyyynmimtuwcq4kbttgt14pbdk3amf7wr9a1cah8ounx953o1zx4gxoj |
+----------------+--------+----------------------+----------------------+-------------------------------------------------------------------+
```
```
> ./safe cat safe://testfolder200820192 --info
Native data type: PublishedSeqAppendOnlyData
Type tag: 1100
XOR name: 0xd9b4e701b8558c2ce915a87d265ba9645542041b7b8de69649cffc5a94cdcc0e


Files of FilesContainer (version 0) at "safe://testfolder200820192":
+----------------+--------+----------------------+----------------------+-------------------------------------------------------------------+
| Name           | Size   | Created              | Modified             | Link                                                              |
+----------------+--------+----------------------+----------------------+-------------------------------------------------------------------+
| /image1.png    | 102400 | 2019-08-20T12:14:26Z | 2019-08-20T12:14:26Z | safe://hbyyyynmimtuwcq4kbttgt14pbdk3amf7wr9a1cah8ounx953o1zx4gxoj |
+----------------+--------+----------------------+----------------------+-------------------------------------------------------------------+
| /text1.txt     | 102400 | 2019-08-20T12:14:26Z | 2019-08-20T12:14:26Z | safe://hbyyyynmimtuwcq4kbttgt14pbdk3amf7wr9a1cah8ounx953o1zx4gxoj |
+----------------+--------+----------------------+----------------------+-------------------------------------------------------------------+
| /webpage1.html | 102400 | 2019-08-20T12:14:26Z | 2019-08-20T12:14:26Z | safe://hbyyyynmimtuwcq4kbttgt14pbdk3amf7wr9a1cah8ounx953o1zx4gxoj |
+----------------+--------+----------------------+----------------------+-------------------------------------------------------------------+
```

**Comment by calumcraig** [comment]
QA verified, moving this to done. Issue #206 found during testing but is not a blocker to this.

**Pull request merged**